app/controllers/packages_controller.rb

```diff
@@ -1,19 +1,36 @@
 class PackagesController < ApplicationController
   before_action :set_event, only: [:index, :show]
   before_action :set_package, only: [:show]
+  before_action :set_photographers_present, only: [:index, :show]
 
-  def index
+   def index
     @packages = Package.all
+    @event_time_slots = (1..12).to_a
+    @unavailable_times = []
+    @event_time_slots.each do |event_time_slot|
+      @time_slot_available = []
```

Comment:
Is there a particular reason you have chosen to use an array for this? A more common (but not necessarily better) approach would be to use a boolean:
        Declare time_slot_available as 'false' upon initialisation and set it to 'true' only if it is not included in the unavailable times. The 'else' statement would not be necessary then.

Reply:
Spot on! I updated the logic! @Matth93